### PR TITLE
Speed up training test critical path

### PR DESCRIPTION
**Pull request by lukstafi**
## What changed

- Shorten `circles_conv` from 1,000 to 750 epochs while retaining the same deterministic CNN, dataset, and final loss-below-0.3 assertion.
- Raise its learning rate from 0.015 to 0.018 to preserve convergence on the shorter schedule.
- Skip the generic 64/128/256/512 GPU block-size seed sweep for this fixed, launch-bound graph while retaining backend-default whole/fission autotune presets.
- Refresh the expected trajectory and runtime notes.

## Why

`circles_conv` was the critical path for `dune test test/training`. Its heavily fissioned SGD step benefits greatly from autotuning, but compiling generic block-size variants added substantial cold-start cost, and the original 5,000-step schedule trained beyond the behavior required by the regression.

The suite-level cold rerun improved from 99.75s to 87.68s on `cc` and from 101.84s to 76.24s on Metal. The classifier still moves from near-random loss (~1.1) to below the existing 0.3 threshold.

The separate Linux oversubscription issue (concurrent Dune tests each using OpenMP) is orthogonal; these reductions remain useful on top of scheduling those tests appropriately.

## Validation

- `OCANNL_BACKEND=cc dune test --force test/training`
- `OCANNL_BACKEND=metal dune test --force test/training`
- `dune build @check`
- `git diff --check`

### CODE REVIEW

**Comment by lukstafi** [comment]
Closing this draft because the change must land in the staging repository first. Superseded by https://github.com/lukstafi/ocannl-staging/pull/242.
